### ISSUE TITLE
labels

### ISSUE THREAD

**Issue by ValentinaMiller** [Glutamat42/Carla-Lane-Detection-Dataset-Generation#2]
Hey,
I wondered whether there is a way to generate masks with the labels. I am trying to train a Neural Network and the input format of the one I'm using is images with their instance and binary segmentation images as labels. I tried it with extern mask generatig but the quality wasn't too good which resulted in a model with low performance. So I'd like to try what is provided when using your git.
I'd appreciate very much is someone has some advice :)
Thanks!

**Comment by juliangebele** [comment]
Hi there,
feel free to use the tool from [Ultra Fast Lane Detection](https://github.com/cfzd/Ultra-Fast-Lane-Detection/blob/master/scripts/convert_tusimple.py), which the authors used for generating binary segmentation masks. We used these tools as well for converting our labels to "binary" segmentation images. The script references every lane index with a color value from 1 - n (n=number of lanes). This tool however can only be used with a total of 4 lanes.  

If you are looking for open source lane detection datasets, feel free to use [our datasets and tools](https://github.com/juliangebele/CARLANE)! 

Kind regards!

**Comment by ValentinaMiller** [comment]
Thank you for the information! However, I can't follow exactly whats happening in the code, could you maybe help me understand it? As far as I understood it, the lanes should be drawn into the image. But I can not see anything, is this encoded into the image somehow?

Thanks !

Maybe it would also be helpful if you could help me understand how the data is saved in the first place as I wonder whether
"Also draws the corresponding ground truth lanepoints on the images and saves the images seperately."
refers to the debug directory or is additionally saved somewhere else?

**Comment by juliangebele** [comment]
The lanes are drawn on the image, but they are barely visible due to their low color value. As mentioned previously, every lane has a unique index, i.e. 1, 2, 3, 4, from left to right, where 1 = the outer left lane, 2 = inner left lane and so on. Now, the index is the acutal color value, which is encoded into the image. You can take a closer look at the draw() method at line 23. Also, you can set `show=True` to better see the lanes. The label images are saved as .png in the same directory of your respective image, which you referenced in the .txt file. After processing, a .txt file is generated, which references the paths of your images and labels as well as some binary information, indicating the presence (1) or absence (0) of the lanes.
